server/controllers/companyController.js

```diff
@@ -92,516 +93,404 @@ const getCompanies = async (req, res) => {
-    if (postRemoveCompanies && postRemoveCompanies.length > 0) {
-      for (const companyName of postRemoveCompanies) {
-        const updateData = {
-          $inc: {},
-        };
-
-        // Construct the dynamic key within inc
-        updateData.$inc[`postCompanies.${companyName.toLowerCase()}`] = -1;
+      //Iterate through and update each next company accordingly
+      for (let i = 0; i < postCompanies.length; i++) {
+        const updateName = postCompanies[i];
+        updateData.$inc[`postCompanies.${updateName}`] = -1;
 
+        // remove self from other company
         const response = await Company.updateOne(
-          { name: lowercaseCompanyName },
-          updateData
+          { name: updateName },
+          { $inc: { [`prevCompanies.${company.name}`]: -1 } }
         );
 
-        if (!response) {
-          res
-            .status(404)
-            .json({ error: `Company not found for ${companyName}` });
-          return;
-        }
+        // Notify when there's an issue updating a company
+        if (!response) console.log(`Error updating ${updateName}`);
       }
-    }
-
-    if (prevRemoveCompanies && prevRemoveCompanies.length > 0) {
-      for (const companyName of prevRemoveCompanies) {
-        const updateData = {
-          $inc: {},
-        };
-
-        updateData.$inc[`prevCompanies.${companyName.toLowerCase()}`] = -1;
-
-        const response = await Company.updateOne(
-          { name: lowercaseCompanyName },
-          updateData
-        );
-
-        if (!response) {
-          res
-            .status(404)
-            .json({ error: `Company not found for ${companyName}` });
-          return;
-        }
+    } else {
+      //Set update rating
+      updateData.$inc["rating"] = company.rating;
+
+      //Set update Tenure
+      let difference = 0;
+      if (!company.indefinite) {
+        const differenceMs =
+          new Date(company.endDate) - new Date(company.startDate); // calc diff in ms
+        // Convert the difference to days
+        difference = math.round(differenceMs / (1000 * 60 * 60 * 24));
+      } else if (
+        company.isIndefinite &&
+        new Date(company.startDate) < new Date()
+      ) {
+        // Calculate the difference in milliseconds
+        const differenceMs = math.abs(date2 - date1);
+        // Convert the difference to days
+        difference = math.round(differenceMs / (1000 * 60 * 60 * 24));
       }
-    }
-
-    if (prevRemoveOtherCompanies && prevRemoveOtherCompanies.length > 0) {
-      for (const companyName of prevRemoveOtherCompanies) {
-        const updateData = {
-          $inc: {},
-        };
-
-        updateData.$inc[`prevCompanies.${lowercaseCompanyName}`] = -1;
+      // Add employees
+      updateData.$inc["tenure"] = difference;
+      updateData.$addToSet["Employees"] = company.userId;
+      updateData.$addToSet["interns"] = company.userId;
+      updateData.$addToSet["ratedEmployees"] = company.userId;
```

Comment:
I'm going to have to start learning this MongoDB magic syntax cause it seems we're going to be using it more.